gems/smithy/spec/support/client_helper.rb

```diff
@@ -0,0 +1,185 @@
+            'integer' => { 'target' => 'smithy.api#Integer' },
+            'list' => { 'target' => 'smithy.ruby.tests#List' },
+            'long' => { 'target' => 'smithy.api#Long' },
+            'map' => { 'target' => 'smithy.ruby.tests#Map' },
+            'short' => { 'target' => 'smithy.api#Short' },
+            'string' => { 'target' => 'smithy.api#String' },
+            'structure' => { 'target' => 'smithy.ruby.tests#Structure' },
+            'timestamp' => { 'target' => 'smithy.api#Timestamp' },
+            'union' => { 'target' => 'smithy.ruby.tests#Union' }
+          }
+        },
+        'smithy.ruby.tests#Union' => {
+          'type' => 'union',
+          'members' => {
+            'string' => { 'target' => 'smithy.api#String' }
+          }
+        }
+      }
+    end
+
+    # @param [Array<String>] module_names A list of module names for the
+    #  generated code. For example, `['Company', 'Weather']` would generate
+    #  code in the `Company::Weather` namespace.
+    # @param [Symbol] type The type of artifact to generate. For example,
+    #  `:schema` or `:client`.
+    # @param [Hash] options Additional options to pass to the generator.
+    # (See Plan#initialize)
+    # @option options [String] :fixture The name of the fixture to load in the
+    #  fixtures folder relative to this file.
+    # @option options [Hash] :model The model to generate code for. Defaults to
+    #  a sample model.
+    # @option options [String] :smithy The smithy version. Defaults to '2.0'.
+    # @option options [Hash] :shapes The shapes to generate code for. Defaults to
+    #  a sample set of shapes.
+    # def sample(module_names, type, options = {})
+    #   model = load_model(options)
+    #   plan = create_plan(module_names, model, type, options)
+    #   if options[:fixture]
+    #     sourced_client(plan)
+    #   else
+    #     generated_client(plan)
+    #   end
+    # end
+
+    def generate(module_names, type, options = {})
+      model = load_fixture(options, module_names)
+      plan_options = {
+        module_name: module_names.join('::'),
+        gem_version: options.fetch(:gem_version, '0.1.0'),
+        destination_root: options.fetch(:destination_root, Dir.mktmpdir),
+        quiet: ENV.fetch('SMITHY_RUBY_QUIET', 'true') == 'true'
+      }
+      plan = Smithy::Plan.new(model, type, plan_options)
+      tmpdir = plan.destination_root
+      Smithy.generate(plan)
+      $LOAD_PATH << ("#{tmpdir}/lib")
+      require plan.gem_name
+      tmpdir
+    end
+
+    def cleanup_generated(module_names, tmpdir = nil)
+      cleanup_sourced(module_names)
+      return unless tmpdir
+
+      if ENV.fetch('SMITHY_RUBY_KEEP_GENERATED_SOURCE', 'false') == 'true'
+        puts "Leaving generated service in: #{tmpdir}"
+      else
+        FileUtils.rm_rf(tmpdir)
+      end
+      $LOAD_PATH.delete("#{tmpdir}/lib")
+    end
+
+    def source(type, options = {})
```

Comment:
I find source a slightly weird name for this since we are doing the eval on it and not returning source.

Reply:
Yes, it's really just a sourced client. I guess this could be "from_source" and "from_files"